build.sh

```diff
@@ -1,110 +1,240 @@
-
-if [ -z ${ghprbActualCommit} ]
+echo "Fetching latest netkan.exe"
+wget --quiet $LATEST_NETKAN_URL -O netkan.exe
+
+# Fetch the latest metadata.
+echo "Fetching latest metadata"
+wget --quiet https://github.com/KSP-CKAN/CKAN-meta/archive/master.tar.gz -O metadata.tar.gz
+
+# Determine KSP dummy name.
+if [ -z $ghprbActualCommit ]
 then
     KSP_NAME=dummy
 else
-    KSP_NAME=${ghprbActualCommit}
+    KSP_NAME=$ghprbActualCommit
 fi
 
-mono --debug ckan.exe ksp add ${KSP_NAME} "`pwd`/dummy_ksp"
-mono --debug ckan.exe ksp default ${KSP_NAME}
-
-echo Running ckan update
-mono --debug ckan.exe update
-
-echo Running jsonlint on the changed files
-echo If you get an error below you should look for syntax errors in the metadata
-
-jsonlint -s -v ${COMMIT_CHANGES}
-
-echo Fetching latest netkan.exe
-
-# fetch latest netkan.exe
-wget --quiet http://ci.ksp-ckan.org:8080/job/NetKAN/lastSuccessfulBuild/artifact/netkan.exe -O netkan.exe
-
-mkdir built
-
-# additional NETKAN_OPTIONS may be set on jenkins jobs
-for f in ${COMMIT_CHANGES}
+# Build all the passed .netkan files.
+# Note: Additional NETKAN_OPTIONS may be set on jenkins jobs
+for f in $COMMIT_CHANGES
 do
-	echo Running NetKAN for $f
-	mono --debug netkan.exe $f --cachedir="dummy_ksp/CKAN/downloads" --outputdir="built" ${NETKAN_OPTIONS}
-done
-
-for f in built/*.ckan
-do
-	echo ----------------------------------------------
-	echo 
-	cat $f | python -m json.tool
-	echo ----------------------------------------------
-	echo 
-	echo Running ckan install -c $f
-	mono --debug ckan.exe install -c $f --headless
+    echo "Running NetKAN for $f"
+    mono netkan.exe $f --cachedir="downloads_cache" --outputdir="built" ${NETKAN_OPTIONS}
 done
 
+# Reset KSP.
+mono ckan.exe ksp forget $KSP_NAME_DEFAULT
 
-echo ----------------------------------------------
-echo
-ADDON_LIST=$(mono ckan.exe list --porcelain)
-OFS=${IFS}
-IFS=' '
-echo ${ADDON_LIST} | while read state name version
+# Test all the built files.
+for f in built/*.ckan
 do
-    # echo "state name version '${state}' '${name}' '${version}'"
-    mono --debug ckan.exe show ${name}
-    echo ----------------------------------------------
-    echo
+    echo "Checking $f"
+    echo "----------------------------------------------"
+    echo ""
+    cat $f | python -m json.tool
+    echo "----------------------------------------------"
+    echo ""
+    
+    # Get a list of all the OTHER files.
+    OTHER_FILES=()
+    
+    for o in built/*.ckan
+    do
+        if [ "$f" != "$o" ]
+        then
+            OTHER_FILES+=($o)
+        fi
+    done
+    
+    # Inject into metadata.
+    inject_metadata $OTHER_FILES
+    
+    # Extract identifier and KSP version.
+    CURRENT_IDENTIFIER=$($JQ_PATH '.identifier' $f)
+    CURRENT_KSP_VERSION=$($JQ_PATH 'if .ksp_version then .ksp_version else .ksp_version_min end' $f)
+    
+    # Strip "'s.
+    CURRENT_IDENTIFIER=$CURRENT_IDENTIFIER//'"'
+    CURRENT_KSP_VERSION=$CURRENT_KSP_VERSION//'"'
+    
```

Comment:
You'll need to put back the curly braces here as the regex doesn't appear to work without them

Without
```
leon@bofh-sider /tmp/NetKAN $ bash /tmp/test.sh 
"blarg"""//"
"""0."00"//"
```
With
```
leon@bofh-sider /tmp/NetKAN $ bash /tmp/test.sh 
blarg
0.00
```

/me learnt something new about bash :-)